aiidalab_widgets_base/nodes.py

```diff
@@ -210,22 +215,30 @@ def _find_called(cls, root):
     @classmethod
     def _find_outputs(cls, root):
         process_node = load_node(root.parent_pk)
-        if root.namespace:
-            outputs = process_node.outputs[root.namespace]
-        else:
-            outputs = process_node.outputs
 
+        outputs = process_node.outputs
+        # if output tree node is from namespace node
+        namespaces = root.namespaces
+        if namespaces:
+            outputs = functools.reduce(
+                lambda d, namespace: d[namespace], namespaces, process_node.outputs
+            )
```

Comment:
I find this segment harder to understand now. Would the following refactoring be correct?
```suggestion
        # Gather outputs from node and its namespaces (if present):
        outputs = functools.reduce(
            lambda d, namespace: d[namespace], root.namespaces or [], process_node.outputs
        )
```